pyomo/contrib/gdpopt/GDPopt.py

```diff
@@ -88,76 +95,55 @@ class GDPoptSolver(object):
     - Logic-to-linear transformation: Romeo Valentin
 
     """
-
-    # Declare configuration options for the GDPopt solver
-    CONFIG = _get_GDPopt_config()
+    CONFIG = ConfigBlock("GDPopt")
+    _add_common_configs(CONFIG)
```

Comment:
They almost could, but `GDPoptSolver` and `_GDPoptAlgorithm` have different `solve` methods: `GDPoptSolver` only exists for backwards compatibility--to support instantiating a solver without specifying the algorithm. Then its `solve` method creates an ephemeral instance of whatever actual solver it needs depending on the algorithm. So I guess we could do a mixin or something, but I kind of like keeping `GDPoptSolver` really minimal because eventually we might just deprecate it and just force people to choose an algorithm?